RedHatInsights/rhsm-subscriptions
# ENT-4384: Create new API for Capacity with metric ID

https://issues.redhat.com/browse/ENT-4384

Test Steps:

- Insert test data: 
```
INSERT INTO public.subscription_capacity VALUES ('account123', 'RHEL', '7025101', 'org123', 20, NULL, false, '2022-06-12 04:00:00+00', '2023-06-11 04:00:00+00', NULL, NULL, 'RH00004', 'Standard', 'Production');
INSERT INTO public.subscription_capacity VALUES ('account123', 'RHEL', '7028601', 'org123', NULL, 2, false, '2022-06-17 04:00:00+00', '2023-06-16 04:00:00+00', NULL, NULL, 'RH00004', 'Standard', 'Production');
INSERT INTO public.subscription_capacity VALUES ('account123', 'RHEL Server', '7025101', 'org123', 20, NULL, false, '2022-06-12 04:00:00+00', '2023-06-11 04:00:00+00', NULL, NULL, 'RH00004', 'Standard', 'Production');
INSERT INTO public.subscription_capacity VALUES ('account123', 'RHEL for x86', '7025101', 'org123', 0, NULL, false, '2022-06-12 04:00:00+00', '2023-06-11 04:00:00+00', 3, NULL, 'RH00004', 'Standard', 'Production');
INSERT INTO public.subscription_capacity VALUES ('account123', 'RHEL for x86', '7028601', 'org123', 0, NULL, false, '2022-06-17 04:00:00+00', '2023-06-16 04:00:00+00', NULL, 30, 'RH00004', 'Standard', 'Production');
INSERT INTO public.subscription_capacity VALUES ('account123', 'RHEL Server', '7028601', 'org123', 0, NULL, false, '2022-06-17 04:00:00+00', '2023-06-16 04:00:00+00', 2, NULL, 'RH00004', 'Standard', 'Production');
```
- Start app: `DEV_MODE=true \ ./gradlew `
- Test endpoint: `
curl -X 'GET' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/capacity/products/RHEL/Sockets?granularity=Daily&beginning=2022-07-21T17%3A32%3A28Z&ending=2022-07-21T17%3A32%3A28Z&category=virtual' \
  -H 'accept: application/vnd.api+json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='`
  - Should only get 2 Virtual sockets and no physical in response.
